### ISSUE TITLE
Player Hand

### ISSUE THREAD

**Issue by I3uckwheat** [I3uckwheat/cards-of-carousal#13]
Blocked By: #11 

This component will be used by the players, mostly on mobile devices, to actually play the game. This feature will have responsibilities such as: selecting white cards, and deselecting white cards. This component should receive a list of cards and have the ability to notify the wrapper of any changes in card selection. This component will also display which cards are selected by using a wrapper. This component is only the functionality of the selecting and deselecting the cards, it does not include the layout and header.

![image](https://user-images.githubusercontent.com/4411701/108461766-f3e0c700-7240-11eb-92cb-6a2a237842eb.png)

 **Don't worry about responsiveness of the white cards for this ticket, that will be handled in another ticket, we just need to make sure the horizontal scrolling works on all screen sizes**

This component should be a "dumb" component, and only respond to events and data, it should not toggle internal state itself when selecting and deselecting cards. That will be the responsibility of the parent.

It will eventually be used similar to the below:
```js
const [selectedCards, setSelectedCards] = useState([])

return (
  <Parent>
    <PlayerHand cards={cards} selected={selectedCards} onSelect={(selected) => setSelectedCards(selected)} />
  </Parent>
)
```

# User Stories
* As a player, I would like to be able to see my hand with all my cards so I can later choose the best cards for the game.
* As a player, I would like to be able to select, and deselect cards before submitting, so I can change my mind before committing to a decision.
* As a developer, I would like the `onSelect` callback to be passed an array of selected card indexes

# Completion Criteria
- [x] Have the following API:
   ```js
   <PlayerHand cards={[Cards]} selected={[Indexes]} onSelect={Function} />
   ```
- [x] Accepts the following data structure for `cards`
   ```js
   [
     "Card One",
     "Card Two",
     "Card Three"
   ]
   ```
- [x] Accepts the following structure for `selectedCards`
   ```js
   [2, 4] // Where 2, and 4 are indexes of selected cards
   ```
- [x] Fires the `onSelect` callback when a card is touched, with the indexes of the selected cards
- [x] Displays the number of the selected card
- [x] Utilizes the card-wrapper component #11 
- [x] Utilizes the white-card component #67 
- [x] Shows the proper order of the cards that have been selected, based on the `selected` array
- [x] Tested
- [x] Uses prop-types
